build/xtask/src/dist.rs

```diff
@@ -1104,9 +1112,26 @@ fn build(
     let prev_time = std::fs::metadata(&src_file).and_then(|f| f.modified());
 
     let status = cmd
-        .status()
+        .output()
         .context(format!("failed to run rustc ({:?})", cmd))?;
-    if !status.success() {
+    std::io::stderr().write_all(&status.stderr);
+    if !status.status.success() {
+        // We've got a special case here: if the kernel memory is too small,
+        // then the build will fail with a cryptic linker error.  We can't
+        // necessarily do normal string search, because the `stderr` may contain
+        // ANSI control characters, so do a *very dumb* search here:
```

Comment:
Is the type-system-level reason here that `status.stderr` is a `Vec<u8>`, and we can't (and wouldn't want to) convert that to a `String`? Could we use https://docs.rs/memchr/latest/memchr/memmem/index.html instead of doing this thing that looks O(n*m) (although `m` is small)?

Reply:
Yeah, we can't convert to a string because there could (and will!) be ANSI control characters in there.  We've already got a `memchr` dep, so I'll switch to that instead.